test/api/pages/delete.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+var _ = require('lodash'),
+  apiAccepts = require('../../fixtures/api-accepts'),
+  endpointName = _.startCase(__dirname.split('/').pop()),
+  filename = _.startCase(__filename.split('/').pop().split('.').shift()),
+  sinon = require('sinon');
+
+describe(endpointName, function () {
+  describe(filename, function () {
+    var sandbox,
+      hostname = 'localhost.example.com',
+      acceptsJson = apiAccepts.acceptsJson(_.camelCase(filename)),
+      acceptsJsonBody = apiAccepts.acceptsJsonBody(_.camelCase(filename)),
+      acceptsHtml = apiAccepts.acceptsHtml(_.camelCase(filename)),
+      pageData = {
+        layout: 'localhost.example.com/components/layout',
+        center: 'localhost.example.com/components/valid',
+        side: ['localhost.example.com/components/valid@valid']
+      },
+      deepData = { deep: {_ref: 'localhost.example.com/components/validDeep'} },
+      layoutData = { someArea: ['center'] },
+      componentData = { name: 'Manny', species: 'cat' };
+
+    beforeEach(function () {
+      sandbox = sinon.sandbox.create();
+      return apiAccepts.beforeEachPageTest(sandbox,  hostname, pageData, layoutData, deepData, componentData);
+    });
+
+    afterEach(function () {
+      sandbox.restore();
+    });
+
+    describe('/pages', function () {
+      var path = this.title;
+
+      acceptsJson(path, {}, 405, { allow:['get', 'post'], code: 405, message: 'Method DELETE not allowed' });
+      acceptsJsonBody(path, {}, {}, 405, { allow:['get', 'post'], code: 405, message: 'Method DELETE not allowed' });
+      acceptsHtml(path, {}, 405, '405 Method DELETE not allowed');
+    });
+
+    describe('/pages/:name', function () {
+      var path = this.title;
+
+      acceptsJson(path, {name: 'valid'}, 200, pageData);
+      acceptsJson(path, {name: 'missing'}, 404, { message: 'Not Found', code: 404 });
```

Comment:
Per slack conversation, there some debate over whether a DELETE should return a 404, since DELETE can be [considered idempotent](https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Idempotent_methods_and_web_applications)

Since we are returning the deleted object (so that the client has the most recent version of what was deleted), then 404 makes sense to me. Essentially, we are forcing a GET prior to DELETE.